app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,368 @@
+  end
+
+  memoize
+  def representer_health
+    # TODO: use error status to determine health (unhealthy if everything fails)
+    return :missing unless track.has_representer?
+    return :needs_attention if representer_display_rate_percentage.zero?
+
+    :exemplar
+  end
+
+  def analyzer
+    {
+      num_runs: Submission::Analysis.joins(submission: :exercise).where(submission: { exercises: { track: } }).count,
+      num_comments: Submission::Analysis.joins(submission: :exercise).where(submission: { exercises: { track: } }).sum(:num_comments),
+      display_rate_percentage: analyzer_display_rate_percentage,
+      volunteers: serialize_tooling_volunteers(track.analyzer_repo_url),
+      health: analyzer_health
+    }
+  end
+
+  memoize
+  def analyzer_display_rate_percentage
+    percentage(Submission::Analysis.with_comments.where(submission: track.submissions).count, track.submissions.count)
+  end
+
+  memoize
+  def analyzer_health
+    # TODO: use error status to determine health (unhealthy if everything fails)
+    return :missing unless track.has_analyzer?
+    return :needs_attention if analyzer_display_rate_percentage.zero?
+    return :healthy if analyzer_display_rate_percentage < 5
+
+    :exemplar
+  end
+
+  def syllabus
+    {
+      concepts:,
+      concept_exercises:,
+      volunteers: syllabus_volunteers,
+      health: syllabus_health
+    }
+  end
+
+  memoize
+  def syllabus_health
+    return :missing if active_concept_exercises.empty?
+    return :needs_attention if active_concept_exercises.size < 10
+    return :needs_attention unless track.course?
+    return :healthy if active_concept_exercises.size < 50
+
+    :exemplar
+  end
+
+  def syllabus_volunteers
+    authors = User.with_attached_avatar.where(id: Concept::Authorship.where(concept: taught_concepts).select(:user_id)).
+      or(User.with_attached_avatar.where(id: Exercise::Authorship.where(exercise: active_concept_exercises).select(:user_id)))
+
+    contributors = User.with_attached_avatar.where(id: Concept::Contributorship.where(concept: taught_concepts).select(:user_id)).
+      or(User.with_attached_avatar.where(id: Exercise::Contributorship.where(exercise: active_concept_exercises).select(:user_id)))
+
+    serialize_volunteers(authors, contributors)
+  end
+
+  def concepts
+    {
+      num_concepts: taught_concepts.size,
+      num_concepts_target:,
+      created: taught_concepts.map { |concept| serialize_concept(concept) }
+    }
+  end
+
+  def num_concepts_target
+    NUM_CONCEPTS_TARGETS.find { |target| taught_concepts.size < target } || taught_concepts.size
+  end
+
+  memoize
+  def taught_concepts = Concept.where(id: concept_taught_exercise.keys).to_a
+
+  memoize
+  def concept_taught_exercise
+    Exercise::TaughtConcept.
+      where(exercise: active_concept_exercises).
+      pluck(:track_concept_id, :exercise_id).
+      map { |concept_id, exercise_id| { concept_id => exercise_id } }.
+      reduce({}, :merge)
```

Comment:
I think you can just do:

```suggestion
      to_h
```

Reply:
I can if I change the above line too:

```ruby
map { |concept_id, exercise_id| [concept_id, exercise_id] }.
      to_h
```